lib/lz4.c

```diff
@@ -865,7 +865,7 @@ LZ4_FORCE_INLINE int LZ4_compress_generic_validated(
     const BYTE* ip = (const BYTE*) source;
 
     U32 const startIndex = cctx->currentOffset;
-    const BYTE* base = (const BYTE*) source - startIndex;
+    const BYTE* base = (const BYTE*) ((uptrval) source - startIndex);
```

Comment:
Documentation states that, by this point, we should have `source != NULL`.

Reply:
`usan` reports [an pointer arithmetic overflow](https://github.com/t-mat/lz4/runs/2683429036?check_suite_focus=true#step:3:100).

```
../lib/lz4.c:868:45: runtime error: pointer index expression with base 0x000001b494b0 overflowed to 0xffffffffffff94b0
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../lib/lz4.c:868:45 in 
```